### PR TITLE
Updating to Microsoft.Azure.ServiceBus 4.1.1. (#26)

### LINKED ISSUE #25: service bus binding does not accept EntityPath as connect string
_From @globysdz on May 26, 2017 18:48_

{
  "disabled": false,
  "bindings": [
    {
      "name" : "myFuncname",
      "topicName" : "request",
      "subscriptionName" : "processor",
      "connection" : "IncomingSubscription"
      "accessRights" : "listen",
      "type" : "serviceBusTrigger",
      "direction" : "in"
    }
  ]
}

the "incomingSubscription" was set via ARM template: 
    {
              "name": "IncomingSubscription",
              "value": "[listkeys(resourceId('Microsoft.ServiceBus/namespaces/topics/authorizationRules', parameters('bus_name'), 'request', 'perm-r'), variables('sbVersion')).primaryConnectionString]" 
     }

the result "inccomingSubscription" string contains ";EntityPath=xxxx" at the end of the string which AzureFunction throws exception at startup time:

Error indexing method 'Functions.globyscadProcessor' ---> System.ArgumentException : ConnectionString used to create MessagingFactory shouldn't include EntityPath as MessagingFactory is at Namespace level.

I am using the same connection string for my webapi application, which works fine. but for function it doesn't work. and there is no good way to automatically set the connection string variable (unless there is a way to obtain the connection string from ARM without EntityPath=xxx part).

_Copied from original issue: Azure/Azure-Functions#315_

### PR DESCRIPTION
**Pull request by alrod**
Merge this after the fix is released:
https://github.com/Azure/azure-sdk-for-net/issues/6410

### CODE REVIEW

File: Message.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="us-ascii"?>
+<Message xmlns="http://schemas.microsoft.com/servicebusexplorer">
+  <Date>Friday, October 4, 2019 10:14:45 PM</Date>
+  <Content><![CDATA[<message>Hi mate, how are you?</message>]]></Content>
+</Message>
```
Comment (brettsam) [inline Message.xml]: What are these Message.xml and Properties.xml files?

File: test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/ServiceBusEndToEndTests.cs [hunk 2]
```diff
@@ -235,6 +235,8 @@ private async Task ServiceBusEndToEndInternal<T>(IHost host = null)
                 string[] consoleOutputLines = consoleOutput
                     .Where(p => p.FormattedMessage != null)
                     .SelectMany(p => p.FormattedMessage.Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries))
+                    // There is a known issue in SB SDK 4.0.0 in MessageReceivePump.MessagePumpTaskAsync https://github.com/Azure/azure-sdk-for-net/issues/6410
+                    .Where(p => !p.StartsWith("Message processing error"))
```
Comment (brettsam) [inline test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/ServiceBusEndToEndTests.cs]: How prevalent is this error? Will this cause a lot of errors logged for customers (and thus CRIs)? Looks like they've fixed it -- should we wait for their next release to move forward?

File: Message.xml (hunk 1 above)
Comment (alrod) [reply Message.xml]: We do not need them, add to PR by mistake. I will remove them.

File: test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/ServiceBusEndToEndTests.cs (hunk 2 above)
Comment (alrod) [reply test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/ServiceBusEndToEndTests.cs]: Make swnse. I will delay the release.

**Pull request merged**